tools/ci.sh

```diff
@@ -117,8 +117,10 @@ function ci_cc3200_build {
 
 function ci_esp32_idf50_setup {
     pip3 install pyelftools
-    git clone https://github.com/espressif/esp-idf.git
+    git clone https://github.com/espressif/esp-idf.git 
```

Comment:
I'm not sure if this is the right way to do it, the migration guide says using `idf.py add-dependency mdns^X.Y.Z` but that only creates a yaml file and the build process does not pull anything from `esp-protocols` and fails since there is no `mdns` component inside `esp-idf/components` ... 🤷🏼‍♂️

Reply:
Yeah, I'm not sure this is the right way to do it.

You should be able to add something to `idf_component.yml` which specifies this component.  Maybe try:
```
espressif/mdns: "~1.1.0"
```

Reply:
I created `ports/esp32/main/idf_component.yaml`  using the command ` idf.py add-dependency "espressif/mdns^1.1.0"
` :
```
## IDF Component Manager Manifest File
dependencies:
  espressif/mdns: "^1.1.0"
  ## Required IDF version
  idf:
    version: ">=4.1.0"
  # # Put list of dependencies here
  # # For components maintained by Espressif:
  # component: "~1.0.0"
  # # For 3rd party components:
  # username/component: ">=1.0.0,<2.0.0"
  # username2/component2:
  #   version: "~1.0.0"
  #   # For transient dependencies `public` flag can be set.
  #   # `public` flag doesn't have an effect dependencies of the `main` component.
  #   # All dependencies of `main` are public by default.
  #   public: true
```
And although the build command detects the dependency, it expects `mdns` component to be already in `esp-idf/components` I guess ?, This is what I get otherwise :
```
CMake Error at esp-idf/tools/cmake/build.cmake:245 (message):
  Failed to resolve component 'mdns'.
Call Stack (most recent call first):
  esp-idf/tools/cmake/build.cmake:281 (__build_resolve_and_add_req)
  esp-idf/tools/cmake/build.cmake:574 (__build_expand_requirements)
  esp-idf/tools/cmake/project.cmake:447 (idf_build_process)
  CMakeLists.txt:53 (project)

```
and 
```
HINT: The component mdns could not be found. This could be because: component name was misspelled, the component was not added to the build, the component has been moved to the IDF component manager or has been removed and refactored into some other component.
Please look out for component in 'https://components.espressif.com' and add using 'idf.py add-dependency' command.
Refer to the migration guide for more details about moved components.
Refer to the build-system guide for more details about how components are found and included in the build.
```
Maybe I'm missing something ? 😵‍💫

Reply:
You don't need to add anything to `esp32_common.mk`.  Just add `main_esp32/idf_component.yml` with the required mdns line and that should be enough.

BUT, when you include mdns.h you need to wrap it like this:
```c
#ifndef NO_QSTR
#include "mdns.h"
#endif
```

Reply:
Thanks that was what I was missing... 😅